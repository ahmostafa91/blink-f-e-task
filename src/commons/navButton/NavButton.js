import React from 'react';
import './NavButton.scss';

const NavButton = ({ buttonIcon }) => {

    return (
        <div className="nav-button">
            {buttonIcon}
        </div>
    );
};

export default NavButton;