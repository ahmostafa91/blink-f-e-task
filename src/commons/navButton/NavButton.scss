.transition {
    -webkit-transition: 0.35s all ease-in-out;
    -moz-transition: 0.35s all ease-in-out;
    -ms-transition: 0.35s all ease-in-out;
    -o-transition: 0.35s all ease-in-out;
    transition: 0.35s all ease-in-out;
    cursor: pointer;
}

.nav-button {
    text-align: center;
    @extend .transition;

    &:hover {
        background-color:rgba(238, 228, 228, 0.322);
    }
}