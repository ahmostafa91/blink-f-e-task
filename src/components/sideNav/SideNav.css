.nav-button, .sideNav-container .log-out, .sideNav-container .nav-options {
  width: 100%;
  color: #fff;
  padding: 25px 0;
}

.sideNav-container {
  display: flex;
  flex-direction: column;
  background-color: #4aafb7;
  height: 100vh;
}

.sideNav-container .logo {
  text-align: center;
  padding: 25px 0;
}

.sideNav-container .logo img {
  width: 3.5em;
}

.sideNav-container .log-out {
  margin-top: auto;
}
