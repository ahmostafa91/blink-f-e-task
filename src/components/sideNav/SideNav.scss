$main-padding: 25px;
.nav-button {
    width: 100%;
    color: #fff;
    padding: $main-padding 0;
  }

.sideNav-container {
  display: flex;
  flex-direction: column;
  background-color: #4aafb7;
  height: 100vh;

  .logo {
    text-align: center;
    padding: $main-padding 0;

    img {
      width: 3.5em;
    }
  }

  .nav-options {
    @extend .nav-button
  }
  .log-out {
    margin-top: auto;
    @extend .nav-button;
  }
}
