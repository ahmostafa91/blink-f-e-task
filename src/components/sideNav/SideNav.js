import React from 'react';
import './SideNav.scss';
import NavButton from '../../commons/navButton/NavButton';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"; 
import { faCog, faSignOutAlt } from "@fortawesome/free-solid-svg-icons";
import { faComments as faCommentsRegular, faUser as faUserRegular } from '@fortawesome/free-regular-svg-icons'

const SideNav = () => {

    return (
      <>
        <section className="sideNav-container">
          <div className="logo">
            <img src={require("../../assets/images/logo_outline.png")} alt="logo" />
          </div>
          <div className="nav-options">
            <NavButton buttonIcon={<FontAwesomeIcon icon={faCommentsRegular} size='3x' />} />
            <NavButton buttonIcon={<FontAwesomeIcon icon={faUserRegular} size='3x' />} />
            <NavButton buttonIcon={<FontAwesomeIcon icon={faCog} size='3x' />} />
            </div>
          <div className="log-out">
            <NavButton buttonIcon={<FontAwesomeIcon icon={faSignOutAlt} size='3x' />} />
          </div>
        </section>
      </>
    );
};

export default SideNav;