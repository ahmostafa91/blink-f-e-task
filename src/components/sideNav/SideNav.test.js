import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import SideNav from './SideNav';
import NavButton from '../../commons/navButton/NavButton';



configure({ adapter: new Adapter() });

describe('<SideNav />', () => {
  const wrapper = shallow(<SideNav />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have four NavButton', () => {
    expect(wrapper.find(NavButton)).toHaveLength(4);
  });

  it('SideNav should have children', () => {
    expect(wrapper.props().children).not.toEqual(null);
  });
  
});
