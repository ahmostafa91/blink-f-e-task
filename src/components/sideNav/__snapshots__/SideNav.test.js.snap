// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav /> should render correctly 1`] = `ShallowWrapper {}`;
